Realm/RLMRealm.h

```diff
@@ -495,6 +496,22 @@ __deprecated_msg("Use `performMigrationForConfiguration:error:`") NS_REFINED_FOR
  */
 + (BOOL)performMigrationForConfiguration:(RLMRealmConfiguration *)configuration error:(NSError **)error;
 
+/**
+ Creates a `RLMThreadHandover` from the passed in objects that can be safely imported on a separate thread.
+
+ Used to pass objects between threads in a thread-safe manner.
+
+ @warning Note that the returned `ThreadHandover` must be imported at most once, and that the current version of the
+ Realm will remain pinned until this package is imported or deallocated.
+
+ @param objects The objects to pass between threads.
+ @return A package that can be imported on a separate thread.
+
+ @see RLMThreadHandover
+ */
+- (RLMThreadHandover *)exportThreadHandoverWithObjects:(NSArray<id<RLMThreadConfined>> *)objects
```

Comment:
A few modifications after further discussion:
The resolve function ought to be defined on `Realm` rather than the `ThreadSafeReference` since it must refresh on lookup, and it's unprecedented to refresh a `Realm` via a method on another object. Also, the resolve function ought to return `nil` if the object no longer exists (instead of returning an invalidated accessor) since it's similar to looking something up by primary key.

```swift
let fooRef = foo.threadSafeReference
let barRef = bar.threadSafeReference
queue.async {
    let realm = try! Realm()
    let foo = realm.resolve(reference: fooRef)! // correct type!
    let bar = realm.resolve(reference: barRef)! // correct type!
    // do stuff
}
```